### PR TITLE
[minor] reduce `substr`

### PR DESCRIPTION
**Pull request by ringabout**

### CODE REVIEW

File: lib/packages/docutils/rstgen.nim [hunk 1]
```diff
@@ -675,7 +675,7 @@ proc readIndexDir(dir: string):
           var x = fileEntries[0].link
           let i = find(x, '#')
           if i > 0:
-            x = x.substr(0, i-1)
+            x.setLen(i-1)
```
Comment (ringabout) [inline lib/packages/docutils/rstgen.nim]: Obviously the range of `i` is 0 .. x.len-1

Comment (Araq) [reply lib/packages/docutils/rstgen.nim]: shouldn't this be:

```nim
x.setLen i
```

?

Comment (ringabout) [inline lib/packages/docutils/rstgen.nim]:
```suggestion
            x.setLen(i)
```

Comment (ringabout) [reply lib/packages/docutils/rstgen.nim]: Yeah, I realized it when I went to sleep yesterday.

**Pull request merged**